### PR TITLE
Deepro dev

### PR DESCRIPTION
**Pull request by TerminalVelocityDPro**
Fixes for motion corruption so that it's properly working within meddlr. Earlier, it turns out motion corruption was just obliterating the image instead of properly motion corrupting it. The specific changes that fixed it are in transform.py - essentially, conversion of image into something that the motion corruption method can handle was missing as well as proper conversion of the image into the kspace using SenseModel and permute. With these fixes, motion corruption does seem to be working properly now.

### CODE REVIEW

File: meddlr/data/transforms/transform.py [hunk 1]
```diff
@@ -22,6 +23,54 @@
 """
 
 
+def add_motion_corruption(
+    image: torch.Tensor,
+    nshots: int,
+    translation: Optional[RandomAffine] = None,
```
Comment (ad12) [inline meddlr/data/transforms/transform.py:29]: rename to `affine_transform`

File: meddlr/data/transforms/transform.py [hunk 2]
```diff
@@ -508,13 +557,19 @@ def __call__(self, kspace, maps, target, fname, slice_id, is_fixed, acceleration
             tfm_gen = RandomAffine(p=1.0, translate=self.translation, angle=self.angle)
             tfm_gen.seed(seed)
 
-            kspace = add_motion_corruption(
+            image = image.permute(0, 3, 1, 2)
```
Comment (ad12) [inline meddlr/data/transforms/transform.py]: Add a comment describing what the shape is here

File: meddlr/data/transforms/transform.py [hunk 3]
```diff
@@ -508,13 +557,19 @@ def __call__(self, kspace, maps, target, fname, slice_id, is_fixed, acceleration
...
+
+            motion_img = add_motion_corruption(
                 image=image,
                 nshots=self.nshots,
                 translation=tfm_gen,
                 trajectory=self.trajectory,
             )
 
+            motion_img = motion_img.permute(0, 2, 3, 1)
```
Comment (ad12) [inline meddlr/data/transforms/transform.py]: Add comment describing shape

File: tools/eval_net.py [hunk 4]
```diff
@@ -218,7 +218,7 @@ def eval(cfg, args, model, weights_basename, criterion, best_value):
         and angle is None
         and translation is None
         and nshots is None
-        and trajectory == "None"
+        and trajectory == "none"
```
Comment (ad12) [inline tools/eval_net.py:222]: why are all of these fields gated by an if statemnt? are you trying to set default values?

File: tools/eval_net.py [hunk 5]
```diff
@@ -512,34 +517,33 @@ def main(args):
...
+        type=float,
         help=("How much rotation angle should be used for motion corruption " "of the dataset"),
     )
 
     parser.add_argument(
         "--translation",
         default=None,
-        type=Optional[float],
+        type=float,
         help=("How much translation should be used for motion " "corruption of the dataset"),
```
Comment (ad12) [inline tools/eval_net.py]: this should be a string, no need to put it in parenthesis

Comment (ad12) [reply tools/eval_net.py]: same for all the variables below

File: tools/eval_net.py [hunk 6]
```diff
@@ -348,7 +353,7 @@ def eval(cfg, args, model, weights_basename, criterion, best_value):
             dataset_name,
             as_test=True,
             add_noise=noise_level > 0,
-            add_motion=include_motion != "false",
+            add_motion=(include_motion != "false" and data_transform is not None),
```
Comment (ad12) [inline tools/eval_net.py]: `include_motion` is a boolean - you dont need to compare it to false.

it should be

```python
add_motion=include_motion and data_transform is not None,
```

File: tools/eval_net.py [hunk 7]
```diff
@@ -303,6 +303,11 @@ def eval(cfg, args, model, weights_basename, criterion, best_value):
             "dataset": dataset_name,
             "Noise Level": noise_level,
             "Motion Level": motion_level,
+            "Angle": str(angle),
+            "Translation": str(translation),
+            "Trajectory": trajectory,
+            "n-shots": str(nshots),
+            "MRI DIM": str(mri_dim),
```
Comment (ad12) [inline tools/eval_net.py]: nit: change to `mri_dim`

**Review by ad12 — CHANGES_REQUESTED** [review]

File: tools/eval_net.py (hunk 4 above)
Comment (TerminalVelocityDPro) [reply tools/eval_net.py:222]: Yes, just setting default values here.

**Pull request merged**